frontend/src/app/modules/challenges/editor-question/editor-question.component.ts

```diff
@@ -54,4 +105,78 @@ export class EditorQuestionComponent implements OnInit {
     public submitQuestion() {
         this.selectedTab = TabType.Help;
     }
+
+    public onNameChange(value: string) {
+        this.nameChange.emit(value);
+
+        this.inputForm.controls.name.setValue(value);
+        this.validationChange.emit(this.inputForm.valid);
+    }
+
+    public onDescriptionChange(value: string) {
+        this.descriptionChange.emit(value);
+
+        this.inputForm.controls.description.setValue(value);
+        this.validationChange.emit(this.inputForm.valid);
+    }
+
+    public onCategoryChange(value: CategoryType) {
+        this.categoryChange.emit(value);
+        this.category = value;
+    }
+
+    public onTagsChange(value: string | string[]) {
+        if (typeof value === 'string') {
+            return;
+        }
+
+        const selectedTags = value.map(tagName =>
+            this.allTags.find(item => item.name === tagName))
+            .filter(item => item !== undefined) as Tag[];
+
+        this.selectedTagsChange.emit(selectedTags);
+
+        this.inputForm.controls.tags.setValue(value);
+        this.validationChange.emit(this.inputForm.valid);
+    }
+
+    public onLevelChange(value: string | string[]) {
+        if (typeof value !== 'string') {
+            return;
+        }
+
+        const selectedLevel = this.allLevels.find(item => item.name === value);
+
+        this.selectedLevelChange.emit(selectedLevel);
+
+        this.inputForm.controls.level.setValue(selectedLevel?.name ?? '');
+        this.validationChange.emit(this.inputForm.valid);
+    }
+
+    public getErrorMessage(formControlName: string) {
+        return getErrorMessage(formControlName, this.inputForm);
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes['checkValidation']) {
+            if (this.checkValidation) {
+                this.inputForm.markAllAsTouched();
+            }
+        }
+        if (changes['allTags']) {
+            this.allTagsNames = this.allTags.map(t => t.name);
+        }
+        if (changes['selectedTags']) {
+            this.selectedTagsNames = this.selectedTags.map(t => t.name);
+            this.inputForm.controls.tags.setValue(this.selectedTagsNames);
+        }
+        if (changes['allLevels']) {
+            this.allLevelsNames = this.allLevels.map(t => t.name);
+        }
+        if (changes['selectedLevel']) {
+            this.selectedLevelName = this.selectedLevel?.name ?? '';
+            this.inputForm.controls.level.setValue(this.selectedLevelName);
+        }
+        fillFormInputs(this.inputForm, changes);
+    }
```

Comment:
I'd extract it to separate methods and do something like 

this.handleValidationChange(changes['checkValidation']);

Reply:
Done